presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -64,11 +65,11 @@ protected Void visitJoin(Join node, MaterializedViewPlanValidatorContext context
                 process(node.getLeft(), context);
                 process(node.getRight(), context);
 
-                context.setProcessingJoinNode(true);
                 if (joinCriteria instanceof JoinOn) {
```

Comment:
This `if` is always true. cc: @gggrace14

Reply:
Yes, @highker is right. Also took a look. @pranjalssh , could you help remove this if-condition in line 68, given it's redundant with line 61. To be more clean, it's not introduced by this PR.